synapse/events/builder.py

```diff
@@ -103,7 +103,7 @@ def is_state(self) -> bool:
 
     async def build(
         self,
-        prev_event_ids: StrCollection,
+        prev_event_ids: List[str],
```

Comment:
#16301 incorrectly changes `Collection[str]` -> `StrSequence`, but even `Collection[str]` is wrong. This needs to be JSON serializable, which pretty much means a `List` or `Tuple`.